biz.aQute.resolve/src/biz/aQute/resolve/Bndrun.java

```diff
@@ -0,0 +1,188 @@
+			Set<Resource> resources = resolution.keySet();
+			List<VersionedClause> runBundles = new ArrayList<>();
+			for (Resource resource : resources) {
+				VersionedClause runBundle = ResourceUtils.toVersionClause(resource, "[===,==+)");
+				if (!runBundles.contains(runBundle)) {
+					runBundles.add(runBundle);
+				}
+			}
+			Collections.sort(runBundles, new Comparator<VersionedClause>() {
+				@Override
+				public int compare(VersionedClause a, VersionedClause b) {
+					int diff = a.getName().compareTo(b.getName());
+					return (diff != 0) ? diff : a.getVersionRange().compareTo(b.getVersionRange());
+				}
+			});
+
+			File runFile = getPropertiesFile();
+			BndEditModel bem = new BndEditModel(getWorkspace());
+			Document doc = new Document(IO.collect(runFile));
+			bem.loadFrom(doc);
+
+			List<VersionedClause> bemRunBundles = bem.getRunBundles();
+			if (bemRunBundles == null)
+				bemRunBundles = new ArrayList<VersionedClause>();
+
+			String originalRunbundlesString = runbundlesWrappedFormatter.convert(bemRunBundles);
+			logger.debug("Original -runbundles was:\n\t {}", originalRunbundlesString);
+			String runbundlesString = runbundlesWrappedFormatter.convert(runBundles);
+			logger.debug("Resolved -runbundles is:\n\t {}", runbundlesString);
+
+			List<VersionedClause> deltaAdd = new ArrayList<>(runBundles);
+			deltaAdd.removeAll(bemRunBundles);
+			List<VersionedClause> deltaRemove = new ArrayList<>(bemRunBundles);
+			deltaRemove.removeAll(runBundles);
+			boolean added = bemRunBundles.addAll(deltaAdd);
+			boolean removed = bemRunBundles.removeAll(deltaRemove);
+			if (added || removed) {
+				if (failOnChanges && !bemRunBundles.isEmpty()) {
+					error("The runbundles have changed. Failing the build!\nWas: %s\nIs: %s", originalRunbundlesString,
+							runbundlesString);
+					return null;
+				}
+				if (writeOnChanges) {
+					bem.setRunBundles(bemRunBundles);
+					String runBundlesProperty = bem.getDocumentChanges().get(Constants.RUNBUNDLES);
+					logger.debug("Writing changes to {}", runFile.getAbsolutePath());
+					logger.debug("{}:{}", Constants.RUNBUNDLES, runBundlesProperty);
+					bem.saveChangesTo(doc);
+					IO.store(doc.get(), runFile);
+				}
+			}
+			return runbundlesListFormatter.convert(bemRunBundles);
+		}
+	}
+
+	/**
+	 * Execute the tests defined by the bndrun configuration.
+	 *
+	 * @param cwd the directory used as the current working directory for the
+	 *            test process
+	 * @param reportsDir the directory in which test reports are written
+	 * @throws Exception
+	 */
+	public int test(File cwd, File reportsDir) throws Exception {
+		logger.info("Test {}", this);
+		ProjectTester projectTester = getProjectTester();
+		cwd.mkdirs();
+		reportsDir.mkdirs();
+
+		projectTester.setCwd(cwd);
+		projectTester.setReportDir(reportsDir);
+
+		return projectTester.test();
+	}
+
+	public void error(AtomicInteger count, String message, Object... args) {
```

Comment:
You will see that bnd-maven-plugin needs its own code to check errors and warnings. You may want to consider using that to play nice with m2e:  https://github.com/bndtools/bnd/blob/master/maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java#L328-L373